proto/src/main/java/com/google/gson/protobuf/ProtoTypeAdapter.java

```diff
@@ -190,6 +191,9 @@ public static Builder newBuilder() {
   private static final com.google.protobuf.Descriptors.FieldDescriptor.Type ENUM_TYPE =
       com.google.protobuf.Descriptors.FieldDescriptor.Type.ENUM;
 
+  private static final ConcurrentMap<String, Map<Class<?>, Method>> mapOfMapOfMethods =
+      new MapMaker().makeMap();
```

Comment:
Does it matter whether it's String -> Map<Class, Method> vs. Class -> Map<String, Method>?  Just wondering since if it could be the latter, you could make the keys weak refs.